### DISCUSSION TITLE
Feature Requests

### DISCUSSION THREAD

**Discussion by vydimitrov** [vydimitrov/react-countdown-circle-timer#67] [Ideas]
## 👋 Welcome!

This is the place to make your feature requests to improve the countdown functionality.

### How does it work?
1. Post your feature request in the comments below
2. Other people will start voting for it by adding a 👍 
3. Once a feature has more than 10 votes it will be added to the countdown

**Comment by dapinitial** [comment]
Clock countdown timer so that I can start it from a certain time and countdown to a certain time! 

Start and stop timer and execute a function.

**Reply by vydimitrov** [reply]
Hey @dapinitial, thanks for posting this feature request. How do you see the component API for this request? You can achieve all of it today with the current API, just that you will need to add a bit of logic, like so:

```jsx
const [isPlaying, setIsPlaying] = useState(true)

 <CountdownCircleTimer
        isPlaying={isPlaying}
        duration={100}
        initialRemainingTime={remainigTime} // here pass the start time
      >
          {({ remainigTime  }) => {
             if (remainigTime === 20) { // here is your end time
                setIsPlaying(false)
               timeReached() // here call your function
            }
            return remainigTime  
          })}
     </CountdownCircleTimer>

```

**Comment by vydimitrov** [comment]
### Gradient directions
Original proposal - https://github.com/vydimitrov/react-countdown-circle-timer/issues/143

**Reply by vydimitrov** [reply]
How about providing a prop that can change the gradient direction? I am not sure how much is the feature used so I do not want  to make some heavy changes just about it.

**Reply by iserzh** [reply]
We can do smth like this:

```
const DIRECTIONS = {
  horizontal: {
    x1: 1,
    y1: 0,
    x2: 0,
    y2: 0,
  },
  'horizontal-inverse': {
    x1: 0,
    y1: 0,
    x2: 1,
    y2: 0,
  },
  'vertical': {
    x1: 0,
    y1: 1,
    x2: 0,
    y2: 0,
  },
  'vertical-inverse': {
    x1: 0,
    y1: 0,
    x2: 0,
    y2: 1,
  },
}

const DefsLinearGradient = ({
  gradientId,
  colors,
  defs: Defs = 'defs',
  linearGradient: LinearGradient = 'linearGradient',
  stop: Stop = 'stop',
  direction: Direction = 'horizontal'
}) => (
  <Defs>
    <LinearGradient id={gradientId} {...DIRECTIONS[direction]}>
      {getStopProps(colors).map((gradient) => (
        <Stop {...gradient} key={gradient.key} />
      ))}
    </LinearGradient>
  </Defs>
)
```

or simply allow user to pass directions as prop with x1, x2, y1, y2 fields

**Reply by vydimitrov** [reply]
Cool, I like the `DIRECTIONS` enum!

**Comment by vydimitrov** [comment]
## Allow no color transition

Make it possible so the color changes instantly without transition.

**Reply by vydimitrov** [reply]
This is now implemented in v3

**Comment by vydimitrov** [comment]
## Return the current color in `onUpdate` handler

This a request to get the current color in `onUpdate` handler as requested in [this issue](https://github.com/vydimitrov/react-countdown-circle-timer/issues/195). One potencial issue I see here is that `onUpdate` is fired once every second so the color value will be changed also once per second.
